packages/@aws-cdk/aws-events/lib/rule.ts

```diff
@@ -413,32 +335,138 @@ export class Rule extends Resource implements IRule {
- * events have the same target, they will share a role.
- * @internal
- */
-  private singletonEventRole(scope: IConstruct, policyStatements: PolicyStatement[]): IRole {
-    const id = 'EventsRole';
-    const existing = scope.node.tryFindChild(id) as IRole;
-    if (existing) { return existing; }
+   * Make sure we add the target environments event bus as a target, and the target has permissions set up to receive our events
+   *
+   * For cross-account rules, uses a support stack to set up a policy on the target event bus.
+   */
+  private ensureXEnvTargetEventBus(targetStack: Stack, targetAccount: string, targetRegion: string, id: string) {
+    // the _actual_ target is just the event bus of the target's account
+    // make sure we only add it once per account per region
+    const key = `${targetAccount}:${targetRegion}`;
+    if (this._xEnvTargetsAdded.has(key)) { return; }
+    this._xEnvTargetsAdded.add(key);
+
+    const eventBusArn = targetStack.formatArn({
+      service: 'events',
+      resource: 'event-bus',
+      resourceName: 'default',
+      region: targetRegion,
+      account: targetAccount,
+    });
 
-    const role = new Role(scope as CoreConstruct, id, {
-      roleName: PhysicalName.GENERATE_IF_NEEDED,
-      assumedBy: new ServicePrincipal('events.amazonaws.com'),
+    // For some region, cross-region requires a Role (with `PutEvents` on the
+    // target event bus) while cross-account doesn't
+    const roleArn = !sameEnvDimension(targetRegion, Stack.of(this).region)
+      ? this.crossRegionPutEventsRole(eventBusArn).roleArn
+      : undefined;
+
+    this.targets.push({
+      id,
+      arn: eventBusArn,
+      roleArn,
     });
 
-    policyStatements.forEach(role.addToPolicy.bind(role));
+    // Add a policy to the target Event Bus to allow the source account/region to publish into it.
+    //
+    // Since this Event Bus permission needs to be deployed before the stack containing the Rule is deployed
+    // (as EventBridge verifies whether you have permissions to the targets on rule creation), this needs
+    // to be in a support stack.
+
+    const sourceApp = this.node.root as App;
+    const sourceAccount = Stack.of(this).account;
+
+    // If different accounts, we need to add the permissions to the target eventbus
+    //
+    // For different region, no need for a policy on the target event bus (but a need
+    // for a role).
+    if (!sameEnvDimension(sourceAccount, targetAccount)) {
+      const stackId = `EventBusPolicy-${sourceAccount}-${targetRegion}-${targetAccount}`;
+      let eventBusPolicyStack: Stack = sourceApp.node.tryFindChild(stackId) as Stack;
+      if (!eventBusPolicyStack) {
+        eventBusPolicyStack = new Stack(sourceApp, stackId, {
+          env: {
+            account: targetAccount,
+            region: targetRegion,
+          },
+          stackName: `${targetStack.stackName}-EventBusPolicy-support-${targetRegion}-${sourceAccount}`,
```

Comment:
`targetRegion` is exactly redundant because it will be deployed there. All `targetRegions` would be the same and hence have no discriminatory value. I guess that might deserve a comment if that's not obvious?

The `sourceRegion` is relevant though.

Also, backwards compat.